10_rest/alextmz/cmd/puppy-server/main.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/anz-bank/go-course/10_rest/alextmz/pkg/puppy"
+	"github.com/anz-bank/go-course/10_rest/alextmz/pkg/puppy/store"
+	"github.com/anz-bank/go-course/10_rest/alextmz/pkg/rest"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	args      = os.Args[1:]
+	flagfile  = kingpin.Flag("data", "JSON file to read").Short('d').Default("./test/valid-formatted-json.json").String()
+	flagport  = kingpin.Flag("port", "TCP port to listen on").Short('p').Default("7735").Uint16()
+	flagstore = kingpin.Flag("store", "Backing store to use").Short('s').Default("sync").Enum("map", "sync")
+
+	out io.Writer = os.Stdout
+
+	// shutdownhttp signals main() to... shutdown the http server
+	shutdownhttp = make(chan bool)
+
+	// syncoutput signals whoever interested that main() output is done
+	syncoutput = make(chan bool, 1)
+)
+
+func main() {
+	if _, err := kingpin.CommandLine.Parse(args); err != nil {
+		fmt.Fprintf(out, "error parsing command line: %v\n", err)
+		return
+	}
+
+	jsonfile, err := os.Open(*flagfile)
```

Comment:
There is a kingpimg flag file type that opens the file as part of command line parsing. saves you the error handling here.